.github/workflows/docker.yml

```diff
@@ -49,35 +40,34 @@ jobs:
           ec2-instance-type: r5.large
           subnet-id: subnet-0983be43
           security-group-id: sg-4cba0d08
-
-      - name: Get EC2 instance meta
-        id: ec2-meta
-        run: |
-          echo "::set-output name=label-${{ matrix.docker-image }}::${{ steps.start-ec2-runner.outputs.label}}"
-          echo "::set-output name=ec2-instance-id-${{ matrix.docker-image }}::${{ steps.start-ec2-runner.outputs.ec2-instance-id}}"
+          aws-resource-tags: >
+            [
+              {"Key": "Name", "Value": "horovod-github-${{ github.head_ref || github.sha }}"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"},
+              {"Key": "GitHubHeadRef", "Value": "${{ github.head_ref }}"},
+              {"Key": "GitHubSHA", "Value": "${{ github.sha }}"}
+            ]
 
   docker:
     name: Build docker image ${{ matrix.docker-image }} (push=${{ github.event_name != 'pull_request' }})
     if: needs.start-runner.result != 'skipped'
     needs: start-runner # required to start the main job when the runner is ready
-    runs-on: ${{ matrix.label }} # run the job on the newly created runners
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runners
 
     # we want an ongoing run of this workflow to be canceled by a later commit
     # so that there is only one concurrent run of this workflow for each branch
     concurrency:
-      group: docker-${{ github.head_ref }}
+      group: docker-${{ matrix.docker-image }}-${{ github.head_ref || github.sha }}
```

Comment:
What does the `github.head_ref || github.sha` expression do? `github.head_ref`, or if empty `github.sha`? That is not documented, is it?

Reply:
Please put your thoughts on this particular expression in a comment here (branch vs. master)

Reply:
Good point. Updated with a comment. And yes, the idea is that on master we want to ensure that every workflow run has a unique ID so we don't cancel any CI runs on master.